### PR TITLE
Mark new API as experimental

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes # 

**Changes proposed in this request**

**Testing**

**Performance impact**

### CODE REVIEW

**Review by SameerK-MSFT — APPROVED** [review]
LGTM

File: src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs [hunk 1]
```diff
@@ -287,11 +288,14 @@ public T WithLegacyCacheCompatibility(bool enableLegacyCacheCompatibility = true
         /// You can set it to <c>true</c> for advanced debugging requiring PII
         /// If both WithLogging apis are set, this one will override the other
         /// </param>
-        /// <returns></returns>
+        /// <returns>The builder to chain the .With methods</returns>
+        /// <item><description>This is an experimental API. The method signature may change in the future without involving a major version upgrade.</description></item>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs:292]: nit:  remarks instead?
```suggestion
        /// <remarks>This is an experimental API. The method signature may change in the future without involving a major version upgrade.</remarks>
```

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/LoggerTests.cs [hunk 2]
```diff
@@ -236,5 +238,22 @@ public async Task IdentityLoggerOverridesLegacyLoggerTestAsync()
                 Assert.IsTrue(testLogger.StringBuilder.ToString().Contains("AcquireTokenByAuthorizationCode"));
             }
         }
+
+        [TestMethod]
+        public void IdentityLoggeExperimental()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/LoggerTests.cs:243]: nit:
```suggestion
        public void IdentityLoggerExperimental()
```

**Review by pmaytak — APPROVED** [review]

**Review by neha-bhargava — APPROVED** [review]

**Comment by SameerK-MSFT** [comment]
The build is successful in [20220620.2](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=940719). Other builds are due to comment changes. Cancelled others to save time.

**Pull request merged**